udata/search/__init__.py

```diff
@@ -22,51 +21,47 @@ def reindex(classname, id):
     document = adapter_class.serialize(obj)
     if adapter_class.is_indexable(obj):
         log.info('Indexing %s (%s)', model.__name__, obj.id)
-        action = KafkaMessageType.INDEX
+        url = f"{current_app.config['SEARCH_SERVICE_API_URL']}{adapter_class.search_url}/index"
+        try:
+            payload = {
+                'document': document
+            }
+            r = requests.post(url, json=payload)
+            r.raise_for_status()
+        except Exception:
+            log.exception('Unable to index/unindex %s "%s"', model.__name__, str(obj.id))
     else:
         log.info('Unindexing %s (%s)', model.__name__, obj.id)
-        action = KafkaMessageType.UNINDEX
-    try:
-        message_type = f'{classname.lower()}.{action.value}'
-        produce(
-            kafka_uri=current_app.config.get('KAFKA_URI'),
-            topic=get_topic(message_type),
-            service='udata',
-            key_id=str(obj.id),
-            document=document,
-            meta={'message_type': message_type, 'index': classname.lower()}
-        )
-    except Exception:
-        log.exception('Unable to index/unindex %s "%s"', model.__name__, str(obj.id))
+        url = f"{current_app.config['SEARCH_SERVICE_API_URL']}{adapter_class.search_url}/{str(obj.id)}/unindex"
+        try:
+            r = requests.delete(url)
+            r.raise_for_status()
+        except Exception:
+            log.exception('Unable to index/unindex %s "%s"', model.__name__, str(obj.id))
 
 
 @task(route='high.search')
 def unindex(classname, id):
     model = db.resolve_model(classname)
+    adapter_class = adapter_catalog.get(model)
     log.info('Unindexing %s (%s)', model.__name__, id)
     try:
-        action = KafkaMessageType.UNINDEX
-        message_type = f'{classname.lower()}.{action.value}'
-        produce(
-            kafka_uri=current_app.config.get('KAFKA_URI'),
-            topic=get_topic(message_type),
-            service='udata',
-            key_id=id,
-            meta={'message_type': message_type, 'index': classname.lower()}
-        )
+        url = f"{current_app.config['SEARCH_SERVICE_API_URL']}{adapter_class.search_url}/{str(id)}/unindex"
+        r = requests.delete(url)
```

Comment:
Does it actually makes sense that we're calling `DELETE` on an endpoint named `unindex`? :p

Reply:
Well If we refer to REST, it deletes a resource, so...